app/models/name.rb

```diff
@@ -408,33 +458,189 @@ class Name < AbstractModel
-        ->(text_name) { where(Name[:text_name].matches("#{text_name} %")) }
+  ### Pattern Search
+  scope :include_synonyms_of,
+        lambda { |name|
+          where(id: name.synonyms.map(&:id)).with_correct_spelling
+        }
+  scope :include_subtaxa_of,
+        lambda { |name|
+          names = [name] + Name.subtaxa_of(name)
+          where(id: names.map(&:id)).with_correct_spelling
+        }
+  scope :text_name_includes,
+        ->(text_name) { where(Name[:text_name].matches("%#{text_name}%")) }
+  scope :with_classification, -> { where(Name[:classification].not_blank) }
+  scope :without_classification, -> { where(Name[:classification].blank) }
+  scope :classification_includes,
+        lambda { |classification|
+          where(Name[:classification].matches("%#{classification}%"))
+        }
+  scope :with_author, -> { where(Name[:author].not_blank) }
+  scope :without_author, -> { where(Name[:author].blank) }
+  scope :author_includes,
+        ->(author) { where(Name[:author].matches("%#{author}%")) }
+  scope :with_citation, -> { where(Name[:citation].not_blank) }
+  scope :without_citation, -> { where(Name[:citation].blank) }
+  scope :citation_includes,
+        ->(citation) { where(Name[:citation].matches("%#{citation}%")) }
+  scope :with_notes, -> { where(Name[:notes].not_blank) }
+  scope :without_notes, -> { where(Name[:notes].blank) }
+  scope :notes_include,
+        ->(notes) { where(Name[:notes].matches("%#{notes}%")) }
+  scope :with_comments, -> { joins(:comments).distinct }
+  scope :without_comments, -> { missing(:comments) }
+  scope :comments_include, lambda { |summary|
+    joins(:comments).where(Comment[:summary].matches("%#{summary}%")).distinct
+  }
+  scope :on_species_list,
+        lambda { |species_list|
+          joins(observations: :species_lists).
+            merge(SpeciesListObservation.where(species_list: species_list))
+        }
+  # Accepts region string, location_id, or Location instance
+  scope :at_location,
+        lambda { |location|
+          case location
+          when String # treat it as a region, not looking for all string matches
+            joins(observations: :location).
+              where(Location[:name].matches("%#{location}"))
+          when Integer, Location
+            joins(:observations).where(observations: { location: location })
+          end
+        }
+  # Occurrence of name within a lat/long box
+  scope :in_box, # Use named parameters (n, s, e, w), any order
+        lambda { |**args|
+          if args[:s].present? && args[:n].present? &&
+             args[:w].present? && args[:e].present? &&
+             (args[:w] < args[:e])
+
+            joins(observations: :location).where(
+              (Location[:south] >= args[:s]).and(Location[:north] <= args[:n]).
+              and(Location[:west] >= args[:w]).and(Location[:east] <= args[:e]).
+              and(Location[:west] <= Location[:east])
+            )
+          end
+        }
```

Comment:
1. Should this be based on the Observations' lat/lon rather their Locations? See Observation.in_box.
2. Do we need this? Could the caller instead chain an Observation scope. Something like
```ruby
Name.<name conditions>.merge(Observation.in_box(args))
```
3. If we do retain it, could it be revised to chain an Observation scope. Something like
```suggestion
  scope :in_box, # Use named parameters (n, s, e, w), any order
        lambda { |args|
          joins(observations: :location).
              merge(Observation.in_box(args)
            )
          end
        }
```